EIPS/eip-7805.md

```diff
@@ -85,31 +85,46 @@ When validators receive ILs from the P2P network, they perform a series of valid
 
 ### Execution Layer
 
-On the execution layer, the block validity conditions are extended such that, after all of the transactions in the block have been executed, we attempt to execute each valid transaction from ILs that was not present in the block.
-If one of those transactions executes successfully, then the block is invalid.
+On the execution layer, an additional check is introduced for new payloads. After all of the transactions in the payload have been executed, we check whether any transaction from ILs, that is not already present in the payload, could be validly included (i.e. nonce and balance checks pass). If that is the case for any transaction, then an error is returned to the CL. Although the block is valid, the CL will not attest to it.
 
 Let `B` denote the current block.
 Let `S` denote the execution state following the execution of the last transaction in `B`.
+Let `gas_left` be the gas remaining after execution of B.
 
 For each transaction `T` in ILs, perform the following:
 
 1. Check whether `T` is present in `B`. If `T` is present, then jump to the next transaction, else continue with next step.
 
-2. Validate `T` against `S`. 
+2. Check whether `B` has enough remaining gas to execute `T`. If `T.gas` > `gas_left`, then jump to the next transaction, else continue with next step.
+
+3. Validate `T` against `S` by checking the nonce and balance of `T.origin`.
```

Comment:
This is clear, but I would mention previously that `S` is thus the state **before** processing the withdrawals (this could lead that the tx sender now has enough balance to execute tx as without processing the withdrawal)

Reply:
The withdrawals happen before transactions execution. See https://ethereum.github.io/consensus-specs/specs/electra/beacon-chain/#block-processing

Reply:
No, withdrawals happen after tx execution, see the EIP:

https://eips.ethereum.org/EIPS/eip-4895#state-transition

> The `withdrawals` in an execution payload are processed **after** any user-level transactions are applied.